packages/kbn-server-http-tools/src/get_listener.test.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  getServerTLSOptionsMock,
+  createHttpServerMock,
+  createHttpsServerMock,
+} from './get_listener.test.mocks';
+import moment from 'moment';
+import { ByteSizeValue } from '@kbn/config-schema';
+import type { IHttpConfig } from './types';
+import { getServerListener } from './get_listener';
+
+const createConfig = (
+  parts: Partial<Omit<IHttpConfig, 'ssl'>> & { ssl?: Partial<IHttpConfig['ssl']> }
```

Comment:
I've tried to simplify this locally, and TS does not complain when doing:
```
parts: Partial<IHttpConfig>
```

Reply:
Pretty sure I had to adapt because partial !== deep partial, and I was using deep partials for `config.ssl`, but I will check again